scipy/ndimage/_morphology.py

```diff
@@ -220,7 +222,14 @@ def _binary_erosion(input, structure, iterations, mask, output,
     except TypeError as e:
         raise TypeError('iterations parameter should be an integer') from e
 
-    input = np.asarray(input)
+    if is_dask(array_namespace(input)):
+        # Note: If you create an dask array with ones
+        # it does a stride trick where it makes an array
+        # (with stride 0) using a scalar
+        # this messes up the C ndimage iteration code
+        input = np.asarray(input, order="C")
```

Comment:
If input needs to be a C-contiguous array, why don't you just pass `order="C"` at all times?
Doesn't this crash if you pass a pure-numpy input that is not C contiguous?
Are we missing a pure-numpy test?

Reply:
Specifying ``order="F`` also works, so the issue is probably with the strides of the array dask makes.
I'm not sure where to debug from here - I tried looking into the C code but couldn't make sense of it.

Since this only affects 3 tests, I'm happy to revert the fix/hack and just skip the tests instead.

Reply:
https://github.com/scipy/scipy/pull/22421
Reverted here.